api/src/api/schemas/search_schema.py

```diff
@@ -0,0 +1,75 @@
+from enum import StrEnum
+from typing import Type
+
+from src.api.schemas.extension import Schema, fields, validators
+
+
+class StrSearchSchemaBuilder:
+    """
+    Builder for setting up a filter in a search endpoint schema.
+
+    Our schemas are setup to look like:
+
+        {
+            "filters": {
+                "field": {
+                    "one_of": ["x", "y", "z"]
+                }
+            }
+        }
+
+    This helps generate the filters for a given field. At the moment,
+    only a one_of filter is implemented.
+
+    Usage::
+
+        # In a search request schema, you would use it like so
+
+        class OpportunitySearchFilterSchema(Schema):
+            example_enum_field = fields.Nested(
+                StrSearchSchemaBuilder("ExampleEnumFieldSchema")
+                    .with_one_of(allowed_values=ExampleEnum)
+                    .build()
+            )
+
+            example_str_field = fields.Nested(
+                StrSearchSchemaBuilder("ExampleStrFieldSchema")
+                    .with_one_of(example="example_value", minimum_length=5)
+                    .build()
+            )
+    """
+
+    def __init__(self, schema_class_name: str):
+        # The schema class name is used on the endpoint
+        self.schema_fields: dict[str, fields.MixinField] = {}
+        self.schema_class_name = schema_class_name
+
+    def with_one_of(
```

Comment:
Is the plan to add `with_all_of` as a filter type as well?

Reply:
We could add that if needed, although I have no idea how the query to the DB would actually work for that.